app_server/app/adapters/socket/Client.js

```diff
@@ -17,7 +17,7 @@ function Client(socket, credentials) {
 
   this.socket = socket;
   this.credentials = credentials;
-  this.rooms = {};
+  this.rooms = {}; // A map from room's name to Room object
```

Comment:
The `Client`'s room only has the rooms that a client a.k.a socket is in. The `rooms` in `RoomsManager` have all rooms.